crates/uv-resolver/src/marker.rs

```diff
@@ -243,7 +259,35 @@ pub(crate) fn normalize_all(tree: MarkerTree) -> Option<MarkerTree> {
             }
         }
 
-        MarkerTree::Expression(_) => Some(tree),
+        MarkerTree::Expression(ref expr) => {
+            if let Some((key, range)) = keyed_range(expr) {
+                // If multiple terms are required to express the range, flatten them into an `Or`
+                // expression.
+                let mut iter = range.iter().flat_map(VersionSpecifier::from_bounds);
+                let first = iter.next().unwrap();
+                if let Some(second) = iter.next() {
```

Comment:
Painful attempt to avoid allocating a vector in the common case (a single term).